packages/app/src/ui/components/VerifiableCredential/Item/VerifiableCredentialItem.tsx

```diff
@@ -0,0 +1,86 @@
+import { VerifiableCredential } from "@cryptkeeperzk/types";
+import CheckIcon from "@mui/icons-material/Check";
+import MoreHorizIcon from "@mui/icons-material/MoreHoriz";
+import IconButton from "@mui/material/IconButton";
+import TextField from "@mui/material/TextField";
+import Typography from "@mui/material/Typography";
+
+import { VerifiableCredentialMetadata } from "@src/types";
+import { Menuable } from "@src/ui/components/Menuable";
+import { ellipsify } from "@src/util/account";
+
+import { useVerifiableCredentialItem } from "./useVerifiableCredentialItem";
+
+export interface VerifiableCredentialItemProps {
+  verifiableCredential: VerifiableCredential;
+  metadata: VerifiableCredentialMetadata;
+  onRenameVerifiableCredential: (hash: string, name: string) => Promise<void>;
+  onDeleteVerifiableCredential: (hash: string) => Promise<void>;
+}
+
+export const VerifiableCredentialItem = ({
+  verifiableCredential,
+  metadata,
+  onRenameVerifiableCredential,
+  onDeleteVerifiableCredential,
+}: VerifiableCredentialItemProps): JSX.Element => {
+  const { isRenaming, name, register, onSubmit, onToggleRenaming, onDelete } = useVerifiableCredentialItem({
+    metadata,
+    onRename: onRenameVerifiableCredential,
+    onDelete: onDeleteVerifiableCredential,
+  });
+
+  const menuItems = [
+    { label: "Rename", isDangerItem: false, onClick: onToggleRenaming },
+    { label: "Delete", isDangerItem: true, onClick: onDelete },
+  ];
+
+  const issuer =
+    typeof verifiableCredential.issuer === "string"
+      ? verifiableCredential.issuer
+      : verifiableCredential.issuer?.id || "unknown";
+
+  return (
+    <div key={metadata.hash} className="p-4 flex flex-row items-center flex-nowrap">
```

Comment:
This should be a Box from material-ui. Take a look at current version of identity list and item, it has some refactoring parts which you can reuse here. This is also related to all the components (Typography, form, icons and etc).